### PR TITLE
Fix select + refactor heist edit/add form

### PR DESCRIPTION
**Pull request by Spacelocust**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #(issue) | No issue linked.

## Type of change

Please add any relevant information to the type of change you've made.

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update (or is a documentation update)
-   [x] HIGH IMPORTANCE (requires a fast merging)

## Has This Been Tested?

Please describe the tests that you ran to verify your changes. It is way more likely for a PR to be merged and approved with successful tests.

-   [x] Yes, I ran tests
-   [ ] No, I did not include any test
-   [ ] No test is needed for this change

### CODE REVIEW

File: front/app/lib/components/form/custom/FieldSelect.tsx [hunk 1]
```diff
@@ -52,34 +42,30 @@ export function FieldSelect<T extends FormData, B extends boolean = false>({
...
-            </>
-          );
-        }}
+        control={control}
+        render={({ field, fieldState: { error } }) => (
+          <>
+            <select
+              {...register(name)}
+              {...rest}
+              aria-describedby={error ? ariaDescribedBy : ''}
```
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldSelect.tsx]:
```suggestion
              aria-describedby={error ? ariaDescribedBy : undefined}
```

File: front/app/lib/components/form/custom/FieldSelect.tsx [hunk 2]
```diff
@@ -1,45 +1,35 @@
...
   hideLabel = false,
   containerClassName = '',
   errorClassName = 'text-accent-6',
   children,
   ...rest
-}: FieldSelectProps<T, OptionFormat, B>) {
+}: FieldSelectProps<T>) {
   const { t } = useTranslation();
   const { control, register } = useRemixFormContext<T>();
   const ariaLabelledBy = `${name}-label`;
```
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldSelect.tsx]: Pas besoin de ça juste mettre un `for` sur le label...

File: front/app/lib/components/form/custom/FieldMultiSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  ...rest
+}: FieldSelectProps<T, OptionFormat, B>) {
+  const { t } = useTranslation();
+  const { control, register } = useRemixFormContext<T>();
+  const ariaLabelledBy = `${name}-label`;
+  const ariaDescribedBy = `${name}-error`;
+
+  return (
+    <Grid className={containerClassName} gap="1">
+      <Text as="span" id={ariaLabelledBy} className={hideLabel ? 'sr-only' : ''}>
```
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldMultiSelect.tsx]: Label + `for`

File: front/app/lib/components/form/custom/FieldMultiSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                (field.value as OptionFormat[]).find((v) => v.value === o.value),
+              )
+            : options?.find((o) => o.value === (field.value as OptionFormat).value);
+          return (
+            <>
+              <Select
+                {...register(name)}
+                {...rest}
+                id={field.name}
+                key={`field_select_key_${JSON.stringify(options)}`}
```
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldMultiSelect.tsx]:
```suggestion
                key={`field_select_key_${JSON.stringify(options)}`}
                aria-describedby={error ? ariaDescribedBy : undefined}
```

Bizarre `key` ?

Comment (Spacelocust) [reply front/app/lib/components/form/custom/FieldMultiSelect.tsx]: La key c'est pour recharger le composant lorsque les options changent, car quand tu changes les options sans la key, le composant ne se recharge pas et les nouvelles options ne sont pas prises en compte

Comment (YummYume) [reply front/app/lib/components/form/custom/FieldMultiSelect.tsx]: :vomiting_face:

**Review by YummYume — APPROVED** [review]

**Pull request merged**